RadClass/RadClass.py

```diff
@@ -63,18 +78,35 @@ def queue_file(self):
 
         Attributes:
         processor: DataSet object responsible for indexing data from file.
-        working_time: The current working epoch timestamp at which data is
-            being analyzed. Used to keep track of progress within a file.
-        start_i: Data index of value working_time. Used in indexing rows
-            for analysis.
+        current_i/start_i: Used in indexing rows for analysis. May be different
+            than the first idx=0 if start_time is specified.
+        stop_i: Last index of timestamps. May be different if stop_time
+            is specified.
         '''
 
         self.processor = ds.DataSet(self.labels)
         self.processor.init_database(self.filename, self.datapath)
         # parameters for keeping track of progress through a file
 
-        self.working_time = self.processor.timestamps[0]
         self.start_i = 0
+        self.stop_i = len(self.processor.timestamps) - 1
+
+        if self.start_time is not None:
+            timestamp = self.processor.timestamps[self.processor.timestamps >=
+                                                  self.start_time][0]
+            self.start_i = np.where(self.processor.timestamps ==
+                                    timestamp)[0][0]
+        else:
+            self.start_time = self.processor.timestamps[0]
```

Comment:
Maybe you can always (re)set `start_time` based on `start_i` as a final way to ensure consistency
```suggestion
        self.start_time = self.processor.timestamps[self.start_i]
```